File: LeetCode/Easy/2293-min-max-game/2293-min-max-game.cpp
class Solution {
public:
    int minMaxGame(vector<int>& nums) {
        if (nums.size() == 1) return nums[0];

        vector<int> temp(nums.size() / 2);
        
        for (int i = 0; i < nums.size() / 2; ++i){
            if (i % 2 == 0) temp[i] = min(nums[i * 2], nums[i * 2 + 1]);
            else temp[i] = max(nums[i * 2], nums[i * 2 + 1]);
        }
                         
        return minMaxGame(temp);
    }
};

File: LeetCode/Easy/2293-min-max-game/2293-min-max-game.py
class Solution:
    def minMaxGame(self, nums: List[int]) -> int:
        while len(nums) > 1:
            temp = [0] * (len(nums) // 2)
            
            i = 0
            is_min = True
            while i < len(nums):
                if is_min:
                    temp[i//2] = min(nums[i], nums[i+1])
                else:
                    temp[i//2] = max(nums[i], nums[i+1])
                is_min = not is_min
                i += 2
            
            nums = temp
        
        return nums[0]